pkg/gds/interceptor.go

```diff
@@ -0,0 +1,28 @@
+package gds
+
+import (
+	"context"
+	"time"
+
+	"github.com/rs/zerolog/log"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (s *Server) serverInterceptor(ctx context.Context, in interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (out interface{}, err error) {
+	// Track how long the method takes to execute.
+	start := time.Now()
+
+	// Check if we're in maintenance mode - status method should still return a full response.
+	if s.conf.Maintenance && info.FullMethod != "/trisa.gds.api.v1beta1.TRISADirectory/Status" {
```

Comment:
@rebeccabilbro made a minor change here so that the `Status` RPC would not return `Unavailable`. Instead, the `Status` RPC will report maintenance mode in a successful response.